### PR TITLE
Rework some configuration fields

### PR DESCRIPTION
**Pull request by lgarciaaco**
- Include ManagementUrlFor3scale and IntegrationStateCheckInterval back in the CR
- Set IntegrationLimit and DeployIntegrations via environment variables
- Include DemoData back into the CR in the root of the CR

Fixes: https://issues.jboss.org/browse/ENTESB-12280

### CODE REVIEW

**Comment by lgarciaaco** [comment]
@avano @mcada I have included your feedback from our previous discussions, can you please have a look?

**Comment by lgarciaaco** [comment]
Fixes https://github.com/syndesisio/syndesis/issues/6356

File: install/operator/pkg/syndesis/configuration/configuration.go [hunk 1]
```diff
@@ -349,12 +350,18 @@ func (config *Config) setConfigFromEnv() error {
 					Image: os.Getenv("DATABASE_IMAGE"), ImageStreamNamespace: os.Getenv("DATABASE_NAMESPACE"),
 					Exporter: ExporterConfiguration{Image: os.Getenv("PSQL_EXPORTER_IMAGE")},
 				},
-				Server: ServerConfiguration{Image: os.Getenv("SERVER_IMAGE")},
+				Server: ServerConfiguration{
+					Image: os.Getenv("SERVER_IMAGE"),
+					Features: ServerFeatures{
+						IntegrationLimit:   setIntFromEnv("INTEGRATION_LIMIT", config.Syndesis.Components.Server.Features.IntegrationLimit),
+						DeployIntegrations: setBoolFromEnv("DEPLOY_INTEGRATIONS", config.Syndesis.Components.Server.Features.DeployIntegrations),
```
Comment (avano) [inline install/operator/pkg/syndesis/configuration/configuration.go]: @lgarciaaco these two should be in CR according to the google doc

File: install/operator/pkg/apis/syndesis/v1alpha1/syndesis_types.go [hunk 2]
```diff
@@ -122,7 +137,20 @@ type VolumeOnlyResources struct {
...
+	IntegrationStateCheckInterval int `json:"integrationStateCheckInterval,omitempty"`
+
+	// Whether we deploy integrations
+	DeployIntegrations bool `json:"deployIntegrations,omitempty"`
+
+	// Set repositories for maven
 	MavenRepositories map[string]string `json:"mavenRepositories,omitempty"`
+
+	// 3scale management URL
+	ManagementUrlFor3scale string `json:"managementUrlFor3scale,omitempty"`
```
Comment (avano) [inline install/operator/pkg/apis/syndesis/v1alpha1/syndesis_types.go]: missing `ControllersIntegrationEnabled` according to the doc

Comment (lgarciaaco) [reply install/operator/pkg/apis/syndesis/v1alpha1/syndesis_types.go]: @avano I don't know which one you mean

Comment (avano) [reply install/operator/pkg/apis/syndesis/v1alpha1/syndesis_types.go]: ah, I see it a bit earlier in the `ServerConfiguration`: https://github.com/syndesisio/syndesis/pull/7153/files#diff-9b395a0be8857358842835be96b2cd70R115

shouldn't that be moved to `ServerFeatures` as well to be consistent?

Comment (lgarciaaco) [reply install/operator/pkg/apis/syndesis/v1alpha1/syndesis_types.go]: @avano I think that property is not used anymore, instead `DeployIntegrations` is used. I am removing it

Comment (avano) [reply install/operator/pkg/apis/syndesis/v1alpha1/syndesis_types.go]: okay

**Comment by lgarciaaco** [comment]
Operator tests passed, I am merging this

**Pull request merged**